lib/lineage.js

```diff
@@ -83,8 +83,75 @@ const fetchHydratedDoc = id => {
   });
 };
 
+const findRow = (rows, id) => id && rows.find(row => row.id === id);
+
+const collectContactIds = docs => {
+  const ids = [];
+  docs.forEach(doc => {
+    const contactId = doc.contact && doc.contact._id;
+    if (contactId && ids.indexOf(contactId) === -1) {
```

Comment:
It's not perfect, but http://node.green is what I use. From that you can see it's been in there since 6.4:
![image](https://user-images.githubusercontent.com/583851/28207586-1ede93e6-6882-11e7-8642-b9d0340eb245.png)